File: backend/src/main/java/com/x/backend/mapper/user/AccountMapper.java
package com.x.backend.mapper.user;


import com.x.backend.pojo.common.Account;
import com.x.backend.pojo.user.dto.account.ForgotPasswordDTO;
import com.x.backend.pojo.user.entity.UserAccount;
import org.apache.ibatis.annotations.Mapper;
import org.springframework.stereotype.Component;

@Component(value = "userAccountMapper")
@Mapper
public interface AccountMapper {
    UserAccount findAccountByUsername(String username);

    Integer validateEmail(String email);

    Integer register(Account account);

    void forgotPassword(ForgotPasswordDTO forgotPasswordDTO);

}

File: backend/src/main/java/com/x/backend/mapper/user/AccountMapper.kt
package com.x.backend.mapper.user

import com.x.backend.pojo.admin.entity.Account
import org.apache.ibatis.annotations.Mapper
import org.springframework.stereotype.Component
import org.springframework.stereotype.Repository
@Component("userAccountMapper")
@Mapper
interface AccountMapper {
    fun findUserByUsername(username:String):Account
}